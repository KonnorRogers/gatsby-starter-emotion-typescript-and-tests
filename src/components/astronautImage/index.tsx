import React from "react";
import { useStaticQuery, graphql } from "gatsby";
import Img, { FluidObject } from "gatsby-image";

type Data = {
  data: {
    placeholderImage: {
      childImageSharp: {
        fluid: FluidObject;
      };
    };
  };
};

/** Refer to https://www.gatsbyjs.org/packages/gatsby-image/#gatsby-image-props for props */
export const PureAstronautImage: React.FC<Data> = ({ data, ...attrs }) => {
  return (
    <Img
      fluid={data.placeholderImage.childImageSharp.fluid}
      alt="An image of an astronaut"
      {...attrs}
    />
  );
};

const AstronautImage: React.FC = ({ ...attrs }) => {
  const data = useStaticQuery(graphql`
    query {
      placeholderImage: file(relativePath: { eq: "gatsby-astronaut.png" }) {
        childImageSharp {
          fluid(maxWidth: 300) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `);

  return <PureAstronautImage data={data} {...attrs} />;
};

export default AstronautImage;
