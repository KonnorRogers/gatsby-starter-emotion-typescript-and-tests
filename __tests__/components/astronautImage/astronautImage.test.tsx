import React from "react";
import { render } from "@testing-library/react";
import data from "Fixtures/astronautImage-fixture";
import { PureAstronautImage as AstronautImage } from "~components/astronautImage";

describe("AstronautImage", () => {
  test("Should render without error", () => {
    const { asFragment, getByAltText } = render(
      <AstronautImage data={data} data-testid="astronaut-image" />
    );

    expect(getByAltText(/An image of an astronaut/)).toBeTruthy();
    expect(asFragment).toMatchSnapshot();
  });
});
