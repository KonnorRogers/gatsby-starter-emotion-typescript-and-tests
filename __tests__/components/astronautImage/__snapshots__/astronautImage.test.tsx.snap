// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstronautImage Should render without error 1`] = `[Function]`;
